app/api/sparql_models.py

```diff
@@ -95,19 +123,24 @@ class Dataset(SPARQLSerializable):
     hasSamples: Subject
     schemaKey: Literal["Dataset"] = "Dataset"
 
-    def to_sparql(self, var_name="?dataset") -> str:
+    def to_sparql(self, var_name: str = "?dataset") -> str:
         cohort_triples = self.to_triples(var_name)
         cohort_triples.extend(
             [f"OPTIONAL {{{var_name} nb:hasPortalURI ?dataset_portal_uri.}}"]
         )
         cohort_triples = "\n    ".join(cohort_triples)
 
         num_sessions_filter = ""
-        if self.hasSamples.hasSession.min_num_imaging_sessions is not None:
+        session = self.hasSamples.hasSession
+        if isinstance(session, PhenotypicSession):
+            min_sessions = session.min_num_phenotypic_sessions
+        else:
```

Comment:
how about we make it explicit here, i.e. `elif isinstance(session, ImagingSession)`?